packages/stark-ui/src/modules/restrict-input-directive/directives/restrict-input.directive.ts

```diff
@@ -25,23 +25,44 @@ export class StarkRestrictInputDirective implements OnInit {
 	 */
 	@HostListener("keypress", ["$event"])
 	public eventHandler(event: KeyboardEvent): boolean {
-		const regularExpression: string = this.inputRestriction || "";
+		// some browsers return the special key value (i.e. keys in the numeric keypad), in such cases we use the 'char'
+		// see: https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key/Key_Values
+		/* tslint:disable-next-line:deprecation */
+		const key: string = event.key.length > 1 ? event.char : event.key;
+
+		return this.testValue(event, key);
+	}
+
+	@HostListener("paste", ["$event"])
+	public onPaste(event: ClipboardEvent): boolean {
+		if (!event.clipboardData) {
+			return true;
+		}
+		const key = event.clipboardData.getData("text/plain");
```

Comment:
Could you rename this to `value` instead of `key` ? 
Since it will probably contains more than one character/key.